### PR TITLE
[Test] Fix and enable test_logging.py

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Fix and enable test_logging.py

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
Can you update the PR descritpion? (just for formality)

File: python/ray/tests/test_logging.py [hunk 1]
```diff
@@ -377,6 +379,10 @@ def f():
     ), f"Python stack trace not found in stderr: {stderr}"
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32" or sys.platform == "darwin",
+    reason="TODO(simon): Failing on Windows and OSX.",
```
Comment (jjyao) [inline python/ray/tests/test_logging.py:384]: @simon-mo I'll leave it to you to fix this: seems java package is not built for windows and osx.
```
(raylet) [2022-03-09 00:41:15,009 C 66657 3215559] (raylet) worker_pool.cc:1316:  Check failed: state != states_by_lang_.end() Required Language isn't supported: JAVA
```

**Comment by rkooo567** [comment]
Waiting for the mac build

**Comment by rkooo567** [comment]
Can you ping me when tests pass?

**Comment by jjyao** [comment]
It's ready to be merged. Failed tests are unrelated.

**Pull request merged**